docs/data/date-pickers/date-picker/date-picker.md

```diff
@@ -98,13 +98,22 @@ You can take advantage of the [PickersDay](/x/api/date-pickers/pickers-day/) com
 
 You can customize the month layout with some props.
 
+### Show additional days
+
 To shows all days of displayed weeks, included those outside of the current month, use `showDaysOutsideCurrentMonth`.
 
 By default, only weeks of the current month are displayed, but you can provide a total number of week to display with `fixedWeekNumber` prop.
 This value is usually set to `6` for Gregorian calendars, because months display can vary between 4 and 6 weeks.
 
 {{"demo": "CustomMonthLayout.js"}}
 
+### Add week number
+
+To display week number, use the `displayWeekNumber`.
+You can customize the string displayed, by using the localization key`localeText.calendarWeekNumberText`.
```

Comment:
I'm unsure if there are use cases where users need to render something different for the week numbers, I'd think there's more chance that they want to update the header.

```suggestion
You can customize the calendar week header by using the localization key `localeText.calendarWeekNumberHeaderText`.
As well you can also customize what's rendered as a calendar Week Number, using a callback for the localization key `localeText.calendarWeekNumberText`
```

Should we also add an example for the callback usage?

Reply:
Not super inspired by what could be a customized week number, so I simply added a point at the end of week numbers, and I override the header to set the default `"#"`. It does not change any thing but it highlight the localization key.